includes/modules/order_total/ot_coupon.php

```diff
@@ -313,23 +313,21 @@ function collect_posts() {
         // JTD - end of handling coupon product restrictions
 
         $date_query=$db->Execute("select coupon_start_date from " . TABLE_COUPONS . "
-                                  where coupon_start_date <= now() and
-                                  coupon_code='" . zen_db_prepare_input($dc_check) . "'");
+                                  where coupon_code='" . zen_db_prepare_input($dc_check) . "' LIMIT 1");
 
-        if ($date_query->RecordCount() < 1 ) {
-          $messageStack->add_session('redemptions', TEXT_INVALID_STARTDATE_COUPON . ' ' . ($dc_link_count ==0 ? $dc_link : ''),'caution');
+        if (date("Y-m-d H:i:s") < $date_query->fields['coupon_start_date']) {
+          $messageStack->add_session('redemptions', sprintf(TEXT_INVALID_STARTDATE_COUPON, ($dc_link_count === 0 ? $dc_link : $dc_check), zen_date_short($date_query->fields['coupon_start_date'])),'caution');
           $error_issues ++;
           $dc_link_count ++;
 //          $this->clear_posts();
 //          zen_redirect(zen_href_link(FILENAME_CHECKOUT_PAYMENT, '', 'SSL', true, false));
         }
 
         $date_query=$db->Execute("select coupon_expire_date from " . TABLE_COUPONS . "
-                                  where coupon_expire_date >= now() and
-                                  coupon_code='" . zen_db_prepare_input($dc_check) . "'");
+                                  where coupon_code='" . zen_db_prepare_input($dc_check) . "'");

```

Comment:
In your other query change for dates you added LIMIT 1, but you didn't do that here. Intentional?

Reply:
no, fixed.